govcd/api_vcd.go

```diff
@@ -68,14 +94,28 @@ func (vcdCli *VCDClient) vcdAuthorize(user, pass, org string) (*http.Response, e
 	req.SetBasicAuth(user+"@"+org, pass)
 	// Add the Accept header for vCA
 	req.Header.Add("Accept", "application/*+xml;version="+vcdCli.Client.APIVersion)
-	resp, err := checkResp(vcdCli.Client.Http.Do(req))
+	resp, err := vcdCli.Client.Http.Do(req)
+
+	// If the VCD has disabled the call to /api/sessions, the attempt will fail with error 401 (unauthorized)
+	// https://docs.vmware.com/en/VMware-Cloud-Director/10.0/com.vmware.vcloud.install.doc/GUID-84390C8F-E8C5-4137-A1A5-53EC27FE0024.html
+	// TODO: convert this method to main once we drop support for 9.7
+	if resp.StatusCode == 401 {
+		resp, err = vcdCli.vcdCloudApiAuthorize(user, pass, org)
+		if err != nil {
+			return nil, err
+		}
+		resp, err = checkRespWithErrType(types.BodyTypeJSON, resp, err, &types.Error{})
+	} else {
+		resp, err = checkResp(resp, err)
+	}
+
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 	// Store the authorization header
-	vcdCli.Client.VCDToken = resp.Header.Get(AuthorizationHeader)
-	vcdCli.Client.VCDAuthHeader = AuthorizationHeader
+	vcdCli.Client.VCDToken = resp.Header.Get(BearerTokenHeader)
+	vcdCli.Client.VCDAuthHeader = BearerTokenHeader
```

Comment:
I'm a bit confused by this part. Have we removed the authorization token completely in favor of the bearer token?

Reply:
No.
We have changed the token that we use to keep the client working. Previously, we injected the authorization token at connection time in the client, so that it would be used for further operations. Now we use the bearer token for the same purpose.

However, if you connect using the old authorization token, it will be used like before.